src/langchain_google_alloydb_pg/pg_to_alloy_migrator.py

```diff
@@ -0,0 +1,529 @@
+        except IndexError as e:
+            raise ValueError(f"Collection: {collection_name}, does not exist.")
+
+    async def _aextract_pgvector_collection(
+        self,
+        collection_name: str,
+        pg_embedding_table_name: Optional[str] = "langchain_pg_embedding",
+        pg_collection_table_name: Optional[str] = "langchain_pg_collection",
+    ) -> Sequence[RowMapping]:
+        """
+        Extract all data belonging to a PGVector collection.
+
+        Args:
+            collection_name (str): The name of the collection to get the data for.
+            pg_embedding_table_name (str): The table name which stores the data corresponding to all collections.
+                Default: "langchain_pg_embedding". Optional.
+            pg_collection_table_name (str): The table name which stores the collection uuid to name mappings.
+                Default: "langchain_pg_collection". Optional.
+
+        Returns:
+            The data present in the collection.
+        """
+        uuid = await self._aget_collection_uuid(
+            collection_name, pg_collection_table_name
+        )
+        try:
+            query = f"SELECT * FROM {pg_embedding_table_name} WHERE collection_id = '{uuid}'"
+            async with self.engine._pool.connect() as conn:
+                result = await conn.execute(text(query))
+                result_map = result.mappings()
+                result_fetch = result_map.fetchall()
+            return result_fetch
+        except:
+            raise ValueError(f"Collection: {collection_name} does not exist.")
+
+    async def _ainsert_single_batch(
+        self,
+        destination_table: str,
+        metadata_column_names: Optional[List[str]],
+        data: Sequence[RowMapping],
+        use_json_metadata: Optional[bool] = False,
+    ) -> None:
+        """
+        Create a batch insert SQL query to insert multiple rows at once.
+        Looks like
+        INSERT INTO MyTable ( Column1, Column2 )
+        VALUES ( Value1, Value2 ), ( Value1, Value2 ), ...
+
+        Args:
+            destination_table (str): The name of the table to insert the data in.
+            metadata_column_names (str): The metadata columns to be created to keep the data in a row-column format.
+                Optional.
+            data (Sequence[RowMapping]): All the data (to be inserted) belonging to a pgvector collection.
+            use_json_metadata (bool): An option to keep the PGVector metadata as json in the AlloyDB table.
+                Default: False. Optional.
+        """
+        row_number = 0
+        params = {}
+
+        if use_json_metadata:
+            insert_query = f"INSERT INTO {destination_table} (langchain_id, content, embedding, langchain_metadata) VALUES"
+
+            # Create value clause for the SQL query
+            values_clause = ", ".join(
+                [
+                    f"(:langchain_id_{num}, :content_{num}, :embedding_{num}, :langchain_metadata_{num})"
```

Comment:
I don't think we need to make an insert statement per row. SQLAlchmey should be able to take an list as the values like:
```python
stmt = "INSERT INTO ... (...) VALUES (:id, :content, ...)"
values = [{id: data.id, content: data.content,...} for row in data]
...
pool.execute(text(stmt), values)

Reply:
Changed.